### PR TITLE
CORE-5247 Combined worker DB bootsptrap

### PR DESCRIPTION
**Pull request by mbrkic-r3**
Fixes related to bootstraping database for combined-worker:
- Fixed Datasource factory selection based on JDBC URL in DbUtils

### CODE REVIEW

File: components/crypto/crypto-db-persistence-model/src/main/kotlin/net/corda/crypto/persistence/db/model/HSMAssociationEntity.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ import javax.persistence.Table
  * Defines a tenant associations for HSMs per category.
  */
 @Entity
-@Table(name = DbSchema.CRYPTO_HSM_ASSOCIATION_TABLE)
+@Table(name = DbSchema.CRYPTO_HSM_ASSOCIATION_TABLE, schema = DbSchema.CRYPTO)
```
Comment (alexey-kadyrov-r3) [inline components/crypto/crypto-db-persistence-model/src/main/kotlin/net/corda/crypto/persistence/db/model/HSMAssociationEntity.kt]: As mentioned in the corda-api the schema has to be part of the connection string

File: testing/db-testkit/src/main/kotlin/net/corda/db/testkit/TestDbInfo.kt [hunk 2]
```diff
@@ -26,15 +28,15 @@ class TestDbInfo(
...
+            .withValue(ConfigKeys.DB_USER, ConfigValueFactory.fromAnyRef("sa"))
+            .withValue(ConfigKeys.DB_PASS, ConfigValueFactory.fromAnyRef(""))
     )
 
-    val emConfig: EntityManagerConfiguration = DbUtils.getEntityManagerConfiguration(
-        inMemoryDbName = name,
-        schemaName = schemaName
-    )
+    // TODO This was temporarily changed to work with in-memory DB until CORE-5418 is fixed
+    val emConfig: EntityManagerConfiguration = TestInMemoryEntityManagerConfiguration(name)
```
Comment (driessamyn) [inline testing/db-testkit/src/main/kotlin/net/corda/db/testkit/TestDbInfo.kt:41]: @charlieR3 FYI
I created https://r3-cev.atlassian.net/browse/CORE-5418 and assigned this to you to have a look at.
This is obviously a hack, but is helping us getting unblocked on getting the all-in-one worker functional.

**Review by driessamyn — COMMENTED** [review]
LGTM
Will merge when api merged.

Green build with alpha version: https://ci02.dev.r3.com/job/Corda5/job/corda-runtime-os/job/PR-1530/11/display/redirect

**Review by driessamyn — DISMISSED** [review]
LGTM

**Review by driessamyn — APPROVED** [review]

**Pull request merged**